### DISCUSSION TITLE
Solar not turning on automatically

### DISCUSSION THREAD

**Discussion by jgagnon5541** [tagyoureit/nodejs-poolController#958] [Nixie]
I trying to get my solar valve to open automatically but it's not working and it remains off even when the solar temp sensor says that it's over 100 degrees.  If I cycle the "power" by pushing the green pool button on the home screen, the solar valve opens as it should and will remain open until the solar temp is withing 2 degrees as set in the heater settings. 
Home screen prior to cycling the green pool power button
![Home Screen](https://github.com/tagyoureit/nodejs-poolController/assets/14060753/5037a1fd-b04c-4523-a649-9213a6755095)

Current settings:
![Heater Settings](https://github.com/tagyoureit/nodejs-poolController/assets/14060753/8763902f-9615-478c-89c7-67e925ea4ff3)
![Valve](https://github.com/tagyoureit/nodejs-poolController/assets/14060753/2c2b7202-200c-4bfb-b14b-f3312cb5ca60)
![Schedule](https://github.com/tagyoureit/nodejs-poolController/assets/14060753/0580a448-c198-404e-aed9-4e3c10283040)

Home screen AFTER I cycle the power button:
![After cycling pool button](https://github.com/tagyoureit/nodejs-poolController/assets/14060753/76c349f6-a741-424d-a903-cc48f5c16097)

I've looked at other posts and believe it's correctly setup but it's simply not turning on automatically.  Anyone have any ideas?

**Comment by rstrouse** [comment]
Are you seeing the device trigger in the REM logs?  What is Solar Valve connected to in REM?

**Reply by jgagnon5541** [reply]
Solar Valve in REM is relay https://github.com/tagyoureit/nodejs-poolController/issues/1 off a sequent 8 relay board. That relay does turn ON in REM and physically when Heater Status shows "Solar" on the home screen. Solar Valve relay https://github.com/tagyoureit/nodejs-poolController/issues/1 is OFF when Heater Status is OFF.

**Reply by rstrouse** [reply]
Do you see any errors in the REM logs?

**Reply by jgagnon5541** [reply]
No errors in REM logs, only many warnings in poolcontroller logs which was issue 245 about longitude and latitude missing.  I've entered the coordinates and that stopped that warning unfortunately it's now too cool to run the solar, don't know if that fixed my solar heater starting automatically unfortunately.

**Reply by jgagnon5541** [reply]
OK even after setting the coordinates it still doesn't start the solar on it's own but this the error I get in the poolcontroller log:
[05/06/2024, 11:27:14] error: Error triggering nixie schedules: Cannot read properties of undefined (reading 'clone')

**Reply by rstrouse** [reply]
Upload a replay.  I will set up a test rig with your configuration.

**Reply by jgagnon5541** [reply]
Not sure how long of a replay you need but this replay starts with the pool in auto, schedule is on, solar temp well above where it needs to start the solar. Had to cycle the pool button to get the solar heater to activate. I can make a longer one if you want starting when the temp is cooler and let it record for a few hours.
[replay_06092024.zip](https://github.com/user-attachments/files/15818359/replay_06092024.zip)

**Reply by jgagnon5541** [reply]
Ok I did another longer replay which cycles through the solar heat starting with much cooler temperature to begin with pump running and the solar heater still doesn't turn on automatically until I turn the pool off and back on using the green button.   Turning the solar heater from solar to off than back to solar does not automatically turn on the solar heater.  Really would like to get this working since at this moment, the only way I have to heat the pool is to manually open the valve.
Also this is a fresh install of Bullseye 32bit using the DIY Standalone Nixie Pool Controller instructions.  I switch to bullseye from bookworm in order to attempt to get my sequant smart fan v6 which unfortunatelly still doesn't work even using the latest REM as of 06202024.
[replay 06212024.zip](https://github.com/user-attachments/files/15929300/replay.06212024.zip)

**Reply by tagyoureit** [reply]
Oof.  This has to do with #925 that was hotly (pun intended) debated 2 months ago.  The original issue had to do with valves short cycling.  The logic for the start and stop temp delta still persists, which may be part of your initial confusion, but now with the changes I just pushed up those will reset with a manual temperature or mode change.

Long term, it would be ideal to put in a solar start/stop temp delay, but those don't exist natively in the OCP's so they weren't added.
